### PR TITLE
Add tests for Number#toExponential

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

**Comment by jugglinmike** [comment]
This patch duplicates assertions in the same was as gh-661, but I'm willing to
let that slide. Just a couple of other things:

- The file named `undefined-fractionDigits.js` should be re-named to use
  lower-case letters
- We're missing an interesting edge case: when the value of "n" is not uniquely
  determined by the algorithm. I had to work this out on a paper, so if you
  need a hand writing that test, I may be able to help.

**Comment by jugglinmike** [comment]
This patch could still use coverage for that edge case.

**Comment by leobalter** [comment]
I missed this one, I'll do it.

**Comment by leobalter** [comment]
I've changed a few assertions to actually compare different results when fractionDigits is undefined and it may act different than being 0.

I don't know if that provides the behavior you asked but it was good to add.

File: test/built-ins/Number/prototype/toExponential/infinity.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2016 The V8 Project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 20.1.3.2
+esid: sec-number.prototype.toexponential
+description: >
+  Return "NaN" if this is NaN
```
Comment (tcare) [inline test/built-ins/Number/prototype/toExponential/infinity.js]: Infinity?

Comment (leobalter) [inline test/built-ins/Number/prototype/toExponential/infinity.js]: oops. fixed

**Comment by tcare** [comment]
Looks good.
